Sources/SwiftSyntax/SymbolName.swift

```diff
@@ -0,0 +1,142 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2014 - 2024 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See https://swift.org/LICENSE.txt for license information
+// See https://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+//===----------------------------------------------------------------------===//
+
+extension FunctionDeclSyntax {
+  /// The symbol name of the function declaration.
+  ///
+  /// The symbol name is a string representation of the function name along with its parameter labels.
+  /// For example, a function `func greet(name: String)` will have the symbol name `greet(name:)`.
+  public var symbolName: String {
+    SwiftSyntax.symbolName(
+      fromBaseName: name.trimmedDescription,
+      andParameters: signature.parameterClause
+    )
+  }
+}
+
+extension InitializerDeclSyntax {
+  /// The symbol name of the initializer declaration.
+  ///
+  /// The symbol name is a string representation of the initializer along with its parameter labels.
+  /// For example, an initializer `init(name: String)` will have the symbol name `init(name:)`.
+  public var symbolName: String {
+    SwiftSyntax.symbolName(
+      // FIXME: If the syntax tree is not generated by parser can this be other token than init?
+      fromBaseName: "init",
+      andParameters: signature.parameterClause
+    )
+  }
+}
+
+extension SubscriptDeclSyntax {
+  /// The symbol name of the subscript declaration.
+  ///
+  /// The symbol name is a string representation of the subscript along with its parameter labels.
+  /// For example, a subscript `subscript(index: Int)` will have the symbol name `subscript(_:)`.
+  public var symbolName: String {
+    SwiftSyntax.symbolName(
+      fromBaseName: "subscript",
+      andParameters: parameterClause
+    )
+  }
+}
+
+extension EnumCaseElementSyntax {
+  /// The symbol name of the enum case element.
+  ///
+  /// The symbol name is a string representation of the enum case name along with its associated value labels (if any).
+  /// For example, an enum case `case foo(bar: Int)` will have the symbol name `foo(bar:)`.
+  public var symbolName: String {
+    let caseName = name.text
+
+    guard let associatedValue = parameterClause else {
+      return caseName
+    }
+
+    let argumentNames = associatedValue.parameters
```

Comment:
Let’s call them `argumentLabels` or `parameterLabels`. A short distinction between arguments and parameters: Parameters are on the function declaration side and declare the things that can be passed to the functions. Arguments are on the call side and specify the values that can be passed to those parameters. The labels are equivalent between arguments and parameters, so we can use either here.

Reply:
`argumentLabels` seems to be the best fit – done.